docs/WindowsCrossCompile.md

```diff
@@ -0,0 +1,47 @@
+# Cross-compiling Swift for Windows with `clang`.
+
+This document describes how to cross compile Swift on Windows on a non-Windows
+host. For more context on the status of Swift on Windows in general, see
+[the Windows doc](./Windows.md)
+
+## 1. Setup Visual Studio Environment Variables
+Building for Windows requires that the Visual Studio environment variables are
+setup similar to the values on Windows. The following assumes that
+`WINKIT_ROOT` points to the path where the Windows 10 SDK is available and that
+`VC_ROOT` points to the path where the Visual Studio VC headers and libraries
+are available. Currently, the runtime has been tested to build against the
+Windows 10 SDK at revision 10.10.586.
+
+```bash
+# Visual Studio 2015 does not have VCToolsInstallDir, use VCINSTALLDIR's value
+export UCRTVersion=10.0.10586.0
+export UniversalCRTSdkDir=".../Windows Kits/10"
+export VCToolsInstallDir=".../Microsoft Visual Studio/2017/Community"
+```
+
+## 2. Setup `visualc` and `ucrt` modules
+The `visualc.modulemap` located at
+`swift/stdlib/public/Platform/visualc.modulemap` needs to be copied into
+`${VCToolsInstallDir}/include`. The `ucrt.modulemap` located at
+`swift/stdlib/public/Platform/ucrt.modulemap` needs to be copied into
+`${UniversalCRTSdkDir}/Include/${UCRTVersion}/ucrt`.
+
+## 3. Configure the runtime to be built with the just built clang
+Ensure that we use the tools from the just built LLVM and clang tools to build
+the Windows SDK. You will need to pass a few extra options to cmake via the
+`build-script` invocation to achieve this. You will need to expand out the
+path where llvm-ar and llvm-ranlib are built. These are needed to correctly
+build the static libraries. Note that cross-compiling will require the use of
+lld. Ensure that lld-link.exe (lld-link) is available to clang via your path.
+Additionally, the ICU headers and libraries need to be provided for the build.
+
+```bash
+--extra-cmake-options=-DSWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER=FALSE, \
+  -DCMAKE_AR=<path to llvm-ar>, \
+  -DCMAKE_RANLIB=<path to llvm-ranlib>, \
+  -DSWIFT_SDKS=WINDOWS, \
+  -DSWIFT_WINDOWS_ICU_I18N_INCLUDE=<path to ICU i18n includes>, \
+  -DSWIFT_WINDOWS_ICU_UC_INCLUDE=<path to ICU UC includes>, \
+  -DSWIFT_WINDOWS_ICU_I18N_LIB=<path to ICU i18n lib>, \
```

Comment:
These lines are outdated - I've been meaning to correct it for a while. The flags are now `-DSWIFT_WINDOWS_ICU_I18N` and `-DSWIFT_WINDOWS_ICU_UC` without the `_LIB`.